lab-doug/lib/app-error.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const debug = require('debug')('matchscore:app-error');
+
+const AppError = module.exports = function(message, statusCode, responseMessage){
+  debug('creating app error');
+  /*
+  inherit the PROPERTIES of Error object.  syntax is method.call(context).  in this example that is Error.call(this)
+  */
+  Error.call(this);
+  this.message = message;
+  this.statusCode = statusCode;
+  this.responseMessage = responseMessage;
+};
+/*
```

Comment:
You write a lot of comments, which is really good!

Something called jsdoc exists, which automatically produces documentation for you based on comments written in your code. You won't need to do that for a long time, but the way that jsdoc comments are formatted is really clear and helpful, and there's a bunch of support out there for it. Even in cases where people have no intention of using JSdoc, a lot of times people will format their comments that way for consistency and readability. 

[This is a package for Atom that I use.](https://atom.io/packages/atom-easy-jsdoc) If you press ctrl+shift+d, it'll insert a pre-formatted comment for you. It'll save you time and make your code look substantially more professional.